torchtext/experimental/datasets/raw/translation.py

```diff
@@ -0,0 +1,391 @@
+
+
+def _clean_xml_file(f_xml):
+    f_txt = os.path.splitext(f_xml)[0]
+    with codecs.open(f_txt, mode='w', encoding='utf-8') as fd_txt:
+        root = ET.parse(f_xml).getroot()[0]
+        for doc in root.findall('doc'):
+            for e in doc.findall('seg'):
+                fd_txt.write(e.text.strip() + '\n')
+
+
+def _clean_tags_file(f_orig):
+    xml_tags = [
+        '<url', '<keywords', '<talkid', '<description', '<reviewer',
+        '<translator', '<title', '<speaker'
+    ]
+    f_txt = f_orig.replace('.tags', '')
+    with codecs.open(f_txt, mode='w', encoding='utf-8') as fd_txt, \
+            io.open(f_orig, mode='r', encoding='utf-8') as fd_orig:
+        for l in fd_orig:
+            if not any(tag in l for tag in xml_tags):
+                # TODO: Fix utf-8 next line mark
+                #                fd_txt.write(l.strip() + '\n')
+                #                fd_txt.write(l.strip() + u"\u0085")
+                #                fd_txt.write(l.lstrip())
+                fd_txt.write(l.strip() + '\n')
+
+
+def _construct_filenames(filename, languages):
+    filenames = []
+    for lang in languages:
+        filenames.append(filename + "." + lang)
+    return filenames
+
+
+def _construct_filepaths(paths, src_filename, tgt_filename):
+    src_path = None
+    tgt_path = None
+    for p in paths:
+        src_path = p if src_filename in p else src_path
+        tgt_path = p if tgt_filename in p else tgt_path
+    return (src_path, tgt_path)
+
+
+def _setup_datasets(dataset_name,
+                    languages,
+                    train_filename,
+                    valid_filename,
+                    test_filename,
+                    root='.data'):
+    src_ext, tgt_ext = languages.split("-")
+    src_train, tgt_train = _construct_filenames(train_filename,
+                                                (src_ext, tgt_ext))
+    src_eval, tgt_eval = _construct_filenames(valid_filename,
+                                              (src_ext, tgt_ext))
+    src_test, tgt_test = _construct_filenames(test_filename,
+                                              (src_ext, tgt_ext))
+
+    extracted_files = []
+    if isinstance(URLS[dataset_name], list):
+        for f in URLS[dataset_name]:
+            dataset_tar = download_from_url(f, root=root)
+            extracted_files.extend(extract_archive(dataset_tar))
+    elif isinstance(URLS[dataset_name], str):
+        dataset_tar = download_from_url(URLS[dataset_name], root=root)
+        extracted_files.extend(extract_archive(dataset_tar))
+    else:
+        raise ValueError(
+            "URLS for {} has to be in a form or list or string".format(
+                dataset_name))
+
+    # Clean the xml and tag file in the archives
+    file_archives = []
+    for fname in extracted_files:
+        if 'xml' in fname:
+            _clean_xml_file(fname)
+            file_archives.append(os.path.splitext(fname)[0])
+        elif "tags" in fname:
```

Comment:
what's in tags file? Do we need to provide them to users?

Reply:
If I understand correctly, this is needed to extract the real content and remove the tags from `.tags` files. This function was there from the branch you created previously, and it was also there in the old translation dataset class.